robottelo/performance/graph.py

```diff
@@ -0,0 +1,97 @@
+import pygal
+
+
+def generate_bar_chart_stat(stat_dict, head, filename, lengend):
+    """Generate Bar chart for stat on concurrent subscription"""
+    bar_chart = pygal.Bar()
+    bar_chart.title = head
+    bar_chart.x_labels = ('min', 'mean', 'max', 'std')
+    bar_chart.x_title = 'Statistics'
+    bar_chart.y_title = 'Time (s)'
+    for key in range(len(stat_dict)):
+        bar_chart.add(
+            '{0}-{1}'.format(lengend, key),
+            stat_dict.get(key)
+        )
+    bar_chart.render_to_file(filename)
+
+
+def generate_line_chart_bucketized_stat(stat_dict, head, filename,
+                                        bucket_size, num_buckets):
+    """Generate Line chart for stat on concurrent subscription"""
+    if bucket_size == 0:
+        return
+
+    # parameters for formats on charts
+    lengend = ('min', 'median', 'max', 'std')
+    buckets = [
+        '{0}-{1}'.format(
+            bucket_size * i + 1,
+            bucket_size * (i + 1)
+        )
+        for i in range(num_buckets)
+    ]
+    line_chart = pygal.Line()
+    line_chart.title = head
+    line_chart.x_labels = buckets
+    line_chart.x_title = 'Buckets'
+    line_chart.y_title = 'Time (s)'
+    for stat, stat_value in enumerate(lengend):
```

Comment:
Yep, you are right for it. I now use `iteritems()` to traverse the dictionary.
Anyway, FYI, the point of code here is to generate line chart for stat values for all clients. For example:
given input of stat on each client, stored as dictionary:
`0: [min, | median,max,std]`
`1: [min, | median,max,std]`
`2: [min, | median,max,std]`
`...`
`9: [min, | median,max,std]`,
it would slice out all min, then all median, all max, and all std sequentially, form 4 lines on the chart, where each line represented by 10 timing points.